drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/test/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/XLSBuilderAttributesTest.java

```diff
@@ -0,0 +1,154 @@
+    }
+
+    @Before
+    public void setUp() throws Exception {
+
+        oldValue = System.getProperty("drools.dateformat");
+
+        System.setProperty("drools.dateformat",
+                           "dd-MMM-yyyy");
+    }
+
+    @After
+    public void tearDown() throws Exception {
+
+        if (oldValue == null) {
+            System.clearProperty("drools.dateformat");
+        } else {
+            System.setProperty("drools.dateformat",
+                               oldValue);
+        }
+    }
+
+    @Test
+    public void headers() {
+
+        assertEquals("RuleSet", cell(1, 1).getStringCellValue());
+        assertEquals("mortgages.mortgages", cell(1, 2).getStringCellValue());
+
+        assertEquals("Import", cell(2, 1).getStringCellValue());
+        assertEquals("", sheet.getRow(2).getCell(2).getStringCellValue());
+
+        assertEquals("RuleTable Attributes", cell(4, 1).getStringCellValue());
+    }
+
+    @Test
+    public void correctAttributeHeaders() {
+
+        assertEquals("PRIORITY", cell(5, 1).getStringCellValue());
+        assertEquals("DATE-EFFECTIVE", cell(5, 2).getStringCellValue());
+        assertEquals("NO-LOOP", cell(5, 3).getStringCellValue());
+        assertEquals("ENABLED", cell(5, 4).getStringCellValue());
+        assertEquals("DATE-EXPIRES", cell(5, 5).getStringCellValue());
+        assertEquals("AGENDA-GROUP", cell(5, 6).getStringCellValue());
+        assertEquals("ACTIVATION-GROUP", cell(5, 7).getStringCellValue());
+        assertEquals("TIMER", cell(5, 8).getStringCellValue());
+        assertEquals("CALENDARS", cell(5, 9).getStringCellValue());
+        assertEquals("AUTO-FOCUS", cell(5, 10).getStringCellValue());
+        assertEquals("LOCK-ON-ACTIVE", cell(5, 11).getStringCellValue());
+        assertEquals("RULEFLOW-GROUP", cell(5, 12).getStringCellValue());
+        assertEquals("DIALECT", cell(5, 13).getStringCellValue());
+        assertEquals("DURATION", cell(5, 14).getStringCellValue());
+    }
+
+    @Test
+    public void patterns() {
+
+        assertNullCell(6, 1);
+        assertNullCell(6, 2);
+        assertNullCell(6, 3);
+    }
+
+    @Test
+    public void constraints() {
+
+        assertNullCell(7, 1);
+        assertNullCell(7, 2);
+        assertNullCell(7, 3);
+        assertNullCell(7, 4);
+        assertNullCell(7, 5);
+        assertNullCell(7, 6);
+        assertNullCell(7, 7);
+        assertNullCell(7, 8);
+        assertNullCell(7, 9);
+        assertNullCell(7, 10);
+        assertNullCell(7, 11);
+        assertNullCell(7, 12);
+        assertNullCell(7, 13);
+        assertNullCell(7, 14);
+    }
+
+    @Test
+    public void columnTitles() {
+
+        assertNullCell(8, 1);
+        assertNullCell(8, 2);
+        assertNullCell(8, 3);
+        assertNullCell(8, 4);
+        assertNullCell(8, 5);
+        assertNullCell(8, 6);
+    }
+
+    @Test
+    public void content() {
+
+        assertEquals("0", cell(9, 1).getStringCellValue());
```

Comment:
Shouldn't be here something like `cell(9, 1).getNumberCellValue()` as the PRIORITY/SALIENCE attribute is number?

Reply:
The attributes are saved as text. No matter what format they are. It could use int, but this works for all.